### ISSUE TITLE
getNickname not working

### ISSUE THREAD

**Issue by Finnlife** [TheHolyWaffle/TeamSpeak-3-Java-API#256]
When i try the getNickname method there is an NullPointerException.
How it works?

**Comment by Henny022** [comment]
maybe the object you are calling the method on is null.
we need some more information, like a code sample, to help you

**Comment by Finnlife** [comment]
When i try this: https://pastebin.com/VZD1jFQW.

When i go in the Channel it dosent work. But Sometimes it works at other Users

**Comment by somefriggnidiot** [comment]
Where is user defined?

**Comment by Finnlife** [comment]
Sorry here: https://pastebin.com/nb53uMKt

**Comment by rogermb** [comment]
Well there's your problem. You're using the UId of the invoker for that lookup, and not the ID of the client who actually moved.

An invoker is an external client that caused that event to happen. For example, the client who kicked / banned a client or who caused that other client to move. (A bit more info on this [here](https://github.com/TheHolyWaffle/TeamSpeak-3-Java-API/wiki/FAQ#what-is-the-difference-between-an-events-client-and-invoker)).

What you're looking for is just `ClientMoveEvent#getClientId()`.

In other words, you should replace
```java
Client user = Load.api.getClientByUId(e.getInvokerUniqueId());
```
with
```java
ClientInfo user = Load.api.getClientInfo(e.getClientId());
```

ClientInfo is just a subclass of Client that contains a bit more information about the client.

Please let us know whether this has solved your issue :smile:

**Comment by Finnlife** [comment]
Ok... thank you. Thats the solved. When i know it its easy. Thank you. Nice Support!

**Comment by rogermb** [comment]
You're welcome! Glad we were able to help! :smile:
